Services/WishListServices.cs

```diff
@@ -0,0 +1,312 @@
+            //check whether the product is in the wishlist already
+            if (!ItemIsInWishlist(wishlist, item)) {
+                //create a new wishlist element and add it
+                var newELement = _contentManager.New<WishListElementPart>("WishListItem");
+                newELement.WishListId = wishlist.ContentItem.Id;
+                newELement.Item = item;
+                _contentManager.Create(newELement.ContentItem);
+                //add to list
+                var elementIds = wishlist.Ids?.ToList();
+                elementIds.Add(newELement.ContentItem.Id);
+                wishlist.Ids = elementIds.ToArray();
+
+                //process extensions
+                foreach (var ext in _wishListExtensionProviders) {
+                    ext.WishListAddedElement(user, wishlist, newELement);
+                }
+            }
+        }
+        public IEnumerable<WishListElementPart> GetElements(IUser user, WishListListPart wishlist) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+            if (wishlist.ContentItem.As<CommonPart>().Owner != user) {
+                throw new ArgumentException(T("Users may only edit wishlists they own.").Text);
+            }
+
+            return GetElements(wishlist);
+        }
+        private IEnumerable<WishListElementPart> GetElements(WishListListPart wishlist) {
+            return _contentManager.Query<WishListElementPart, WishListElementPartRecord>()
+                .Where(epr => epr.WishListId == wishlist.ContentItem.Id)
+                .List();
+        }
+
+        public bool ItemIsInWishlist(WishListListPart wishlist, ShoppingCartItem item) {
+            return GetElements(wishlist).Any(wel =>
+                ShoppingCartItem.ItemsAreEqual(wel.Item, item));
+        }
+
+        public void RemoveElementFromWishlist(IUser user, WishListListPart wishlist, int elementId) {
+            if (user == null) {
+                throw new ArgumentNullException("user");
+            }
+            if (wishlist.ContentItem.As<CommonPart>().Owner != user) {
+                throw new ArgumentException(T("Users may only edit wishlists they own.").Text);
+            }
+
+            var elementPart = _contentManager.Get<WishListElementPart>(elementId);
+            
+            RemoveElementFromWishlist(wishlist, elementPart);
+        }
+        private void RemoveElementFromWishlist(WishListListPart wishlist, WishListElementPart elementPart) {
+            //process extensions
+            foreach (var ext in _wishListExtensionProviders) {
+                ext.WishListElementCleanup(wishlist, elementPart);
+            }
+
+            var elementId = elementPart.ContentItem.Id;
+            if (wishlist.Ids.Contains(elementId)) {
+                var elementIds = wishlist.Ids.ToList().Where(id => id != elementId);
+                wishlist.Ids = elementIds.ToArray();
```

Comment:
The `ToList` is completely useless there.
The `ToArray` is because the property I am assigning to is an `int []`